Pcap++/header/PcapLiveDevice.h

```diff
@@ -406,17 +406,39 @@ namespace pcpp
 		 */
 		bool captureActive();
 
+		/**
+		 * Checks whether the packetPayloadLength is larger than the device MTU. Logs an error if check fails
+		 * @param[in] packetPayloadLength The length of the IP layer of the packet
+		 * @return True if the packetPayloadLength is less than or equal to the device MTU
+		 */
+		bool doMtuCheck(int packetPayloadLength);
+
 		/**
 		 * Send a RawPacket to the network
 		 * @param[in] rawPacket A reference to the raw packet to send. This method treats the raw packet as read-only, it doesn't change anything
 		 * in it
+		 * @param[in] checkMtu Whether the length of the packet's payload should be checked against the MTU. If enabled this comes with a small performance penalty.
 		 * @return True if packet was sent successfully. False will be returned in the following cases (relevant log error is printed in any case):
 		 * - Device is not opened
 		 * - Packet length is 0
 		 * - Packet length is larger than device MTU
 		 * - Packet could not be sent due to some error in libpcap/WinPcap/Npcap
 		 */
-		bool sendPacket(RawPacket const& rawPacket);
+		bool sendPacket(RawPacket const& rawPacket, bool checkMtu = false);
+
+		/**
+		 * Send a buffer containing packet raw data (including all layers) to the network
+		 * @param[in] packetData The buffer containing the packet raw data
+		 * @param[in] packetDataLength The length of the buffer (this is the entire packet, including link layer)
+		 * @param[in] packetPayloadLength The length of the payload for the data link layer. This includes all data apart from the header for the
+		 * data link layer.
+		 * @return True if the packet was sent successfully. False will be returned in the following cases (relevant log error is printed in any case):
+		 * - Device is not opened
+		 * - Packet data length is 0
+		 * - Packet payload length is larger than device MTU
+		 * - Packet could not be sent due to some error in libpcap/WinPcap/Npcap
+		 */
+		bool sendPacket(const uint8_t* packetData, int packetDataLength, int packetPayloadLength);
```

Comment:
This method seems a bit confusing to me. I'm not sure users will really understand the difference between `packetDataLength` and `packetPayloadLength` (although it's explained very well in the doxygen documentation).
Also, how should users "guess" the `packetPayloadLength` without parsing the packet?

Reply:
Given that this method takes a byte array rather than a `pcpp::Packet` or `pcpp::RawPacket` I thought it would be useful to be able to run the check for the MTU without incurring the penalty of packet parsing if you happened to have this information already. For example, in my own use case, I was using PcapPlusPlus to send packets produced by a network simulator. Given that the simulator had its own `packet` representation, it was possible to find this info without additional parsing. It is probably quite a niche use case where users would already have access to this information though.

If you would prefer to remove this method, considering it may be confusing, I can do that; or I could add a bit more to the documentation explaining when it should be used?

Also, for the method:
```c++
bool sendPacket(const uint8_t* packetData, int packetDataLength)
```
it made sense in my head to have this method not parse the MTU, and to have
```c++
bool sendPacket(const uint8_t* packetData, int packetDataLength, int packetPayloadLength)
```
for parsing the MTU, in order to keep the implementations which used byte arrays simple. In hindsight, in order to have more intuitive function signatures, it is probably worth including the default parameters:
```c++
bool checkMtu = false, pcpp::LinkLayerType linkType = pcpp::pcpp::LINKTYPE_ETHERNET
```
which were mentioned earlier in the thread #598, to give the option of parsing the packets and checking MTU. I'll add that back in unless there's any reason not to.

Reply:
Yes that makes sense. Let's keep that method and add a little more documentation